src/main/java/com/fasterxml/jackson/databind/introspect/EnumNamingStrategyFactory.java

```diff
@@ -33,10 +33,34 @@ private EnumNamingStrategyFactory() {}
      *                                  not a subclass of {@link EnumNamingStrategy}.
      *                                  
      * @since 2.15
+     * @deprecated Since 2.19, use variant that takes {@link EnumNamingStrategy} instead.
      */
+    @Deprecated
     public static EnumNamingStrategy createEnumNamingStrategyInstance(Object namingDef, boolean canOverrideAccessModifiers) {
+        return createEnumNamingStrategyInstance(namingDef, canOverrideAccessModifiers, null);
+    }
+
+    /**
+     * Factory method for creating an instance of {@link EnumNamingStrategy} from a provided {@code namingDef}.
+     *
+     * @param namingDef                  subclass of {@link EnumNamingStrategy} to initialize an instance of.
+     * @param canOverrideAccessModifiers whether to override access modifiers when instantiating the naming strategy.
+     * @param defaultNamingStrategy      configured global {@link EnumNamingStrategy} to use in case {@code namingDef} is not provided.
+     *
+     * @return an instance of {@link EnumNamingStrategy} if {@code namingDef} is a subclass of {@link EnumNamingStrategy},
+     * {@code null} if {@code namingDef} is {@code null},
+     * and an instance of {@link EnumNamingStrategy} if {@code namingDef} already is one.
+     *
+     * @throws IllegalArgumentException if {@code namingDef} is not an instance of {@link java.lang.Class} or
+     *                                  not a subclass of {@link EnumNamingStrategy}.
+     *
+     * @since 2.19
+     */
+    public static EnumNamingStrategy createEnumNamingStrategyInstance(Object namingDef,
+                                                                      boolean canOverrideAccessModifiers,
+                                                                      EnumNamingStrategy defaultNamingStrategy) {
         if (namingDef == null) {
-            return null;
+            return defaultNamingStrategy;
```

Comment:
This behavior does not match the documentation wrt null handling. You are already passing `null` from the deprecated API `createEnumNamingStrategyInstance`

Reply:
Thanks, corrected the doc:
```java
     * @return an instance of {@link EnumNamingStrategy} if {@code namingDef} is a subclass of {@link EnumNamingStrategy},
     * {@code defaultNamingStrategy} if {@code namingDef} is {@code null},
     * and an instance of {@link EnumNamingStrategy} if {@code namingDef} already is one.

```